src/VirtoCommerce.CustomerExportImportModule.Data/Validation/ImportAccountValidator.cs

```diff
@@ -0,0 +1,104 @@
+using System.Collections.Generic;
+using System.Linq;
+using FluentValidation;
+using Microsoft.AspNetCore.Identity;
+using VirtoCommerce.CustomerExportImportModule.Core.Models;
+using VirtoCommerce.CustomerExportImportModule.Data.Helpers;
+using VirtoCommerce.Platform.Core;
+using VirtoCommerce.Platform.Core.Security;
+using VirtoCommerce.Platform.Core.Settings;
+using VirtoCommerce.StoreModule.Core.Model.Search;
+using VirtoCommerce.StoreModule.Core.Services;
+
+namespace VirtoCommerce.CustomerExportImportModule.Data.Validation
+{
+    public class ImportAccountValidator : AbstractValidator<ImportRecord<ImportableContact>>
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly IStoreSearchService _storeSearchService;
+        private readonly ISettingsManager _settingsManager;
+
+        public ImportAccountValidator(UserManager<ApplicationUser> userManager, IStoreSearchService storeSearchService, ISettingsManager settingsManager)
+        {
+            _userManager = userManager;
+            _storeSearchService = storeSearchService;
+            _settingsManager = settingsManager;
+            AttachValidators();
+        }
+
+        private void AttachValidators()
+        {
+            When(x => new[]
+                {
+                    x.Record.AccountId, x.Record.AccountType, x.Record.AccountStatus, x.Record.AccountLogin, x.Record.AccountEmail, x.Record.StoreId, x.Record.StoreName,
+                    x.Record.EmailVerified.ToString()
+                }.Any(field => !string.IsNullOrEmpty(field))
+                , () =>
+                {
+
+                    RuleFor(x => x.Record.AccountLogin)
+                        .NotEmpty()
+                        .WithMissingRequiredValueCodeAndMessage("Account Login")
+                        .WithImportState()
+                        .DependentRules(() =>
+                        {
+                            RuleFor(x => x.Record.AccountLogin)
+                                .MustAsync(async (_, userName, __) => await _userManager.FindByNameAsync(userName) == null)
```

Comment:
Comparing with task is unexpected for me here. But okay. :)

Reply:
I used `()` first and was confused when R# said it's useless because it will be compared with result after `await`, but seems like it's true